### PR TITLE
Volumes

### PR DESCRIPTION
**Pull request by V-Gorash**
### WHY
Volumes are not supported by hdRPR

### WHAT
Now VDB volumes can be rendered using Northstar

### HOW
There was an old volumes code, but it was incompatible with Northstar API. Now API calls are changed to be compatible with new Northstar API

### NOTES
Only VDB volumes are supported now

### CODE REVIEW

**Comment by ARozghon** [comment]
waiting for the fix for the Temperature parameters

**Comment by ARozghon** [comment]
Approved by QA

File: pxr/imaging/plugin/hdRpr/volume.cpp [hunk 1]
```diff
@@ -406,52 +406,70 @@ void HdRprVolume::Sync(
...
+            if (desc.fieldName == HdRprVolumeTokens->density) {
+                processVdbGridInfo(densityGridInfo, desc);
+                densityGridFound = true;
+            }
+            // now processing of temperature grid is temporary disabled because it can produce incorrect result. It will be re-implemented in the future using volume material 
+             
+            /* else if (desc.fieldName == HdRprVolumeTokens->temperature) {
+                processVdbGridInfo(emissionGridInfo, desc);
+                bool emissionGridFound = true;
+            }*/
```
Comment (dmitrii-z) [inline pxr/imaging/plugin/hdRpr/volume.cpp:463]: Please remove commented code

File: pxr/imaging/plugin/hdRpr/rprApi.cpp [hunk 2]
```diff
@@ -334,12 +334,19 @@ class HdRprApiRawMaterial : public RprUsdMaterial {
...
 
 struct HdRprApiVolume {
-    std::unique_ptr<rpr::HeteroVolume> heteroVolume;
-    std::unique_ptr<rpr::Grid> albedoGrid;
     std::unique_ptr<rpr::Grid> densityGrid;
-    std::unique_ptr<rpr::Grid> emissionGrid;
-    std::unique_ptr<rpr::Shape> cubeMesh;
-    std::unique_ptr<HdRprApiRawMaterial> cubeMeshMaterial;
+
+    std::unique_ptr<rpr::Shape> base_mesh;
```
Comment (dmitrii-z) [inline pxr/imaging/plugin/hdRpr/rprApi.cpp:339]: Let's be consistent and name it baseMesh, in the style of other fields.

File: pxr/imaging/plugin/hdRpr/rprApi.cpp [hunk 3]
```diff
@@ -1466,69 +1473,96 @@ class HdRprApiImpl {
...
+                continue;
+            }
+            if (densityValues[idx] >= 1) {
+                computedDensityValues[idx] = densityLUT.back()[0];
+                continue;
+            }
+            size_t lookupIndex = floor(densityValues[idx] * (densityLUT.size() - 1));
+            
+            //linear interpolation
+            float firstValue = densityLUT[lookupIndex][0];
```
Comment (dmitrii-z) [inline pxr/imaging/plugin/hdRpr/rprApi.cpp:1502]: Is that guaranteed that lookupIndex is within the vector bounds?

File: pxr/imaging/plugin/hdRpr/rprApi.cpp [hunk 4]
```diff
@@ -3909,6 +3939,29 @@ Don't show this message again?
         return CreateMesh(position, indexes, normals, VtIntArray(), VtVec2fArray(), VtIntArray(), vpf);
     }
 
+    rpr::Shape* CreateVoidMesh() { // creates special mesh which can be used as a base for hetero volumes
+
+        rpr_mesh_info mesh_properties[16];
```
Comment (dmitrii-z) [inline pxr/imaging/plugin/hdRpr/rprApi.cpp:3944]: again, meshProperties seems more consistent

File: pxr/imaging/plugin/hdRpr/rprApi.cpp (hunk 3 above)
Comment (V-Gorash) [reply pxr/imaging/plugin/hdRpr/rprApi.cpp:1502]: Yes because we check index in lines 1490-1498

**Pull request merged**